pkg/volume/volume_linux.go

```diff
@@ -63,32 +65,21 @@ func SetVolumeOwnership(mounter Mounter, fsGroup *int64, fsGroupChangePolicy *v1
 		if err != nil {
 			return err
 		}
-		return changeFilePermission(path, fsGroup, mounter.GetAttributes().ReadOnly, info)
+		return changeFilePermission(path, fsUser, fsGroup, mounter.GetAttributes().ReadOnly, info)
 	})
 
 }
 
-func legacyOwnershipChange(mounter Mounter, fsGroup *int64) error {
+func legacyOwnershipChange(mounter Mounter, fsUser, fsGroup *int64) error {
 	return filepath.Walk(mounter.GetPath(), func(path string, info os.FileInfo, err error) error {
 		if err != nil {
 			return err
 		}
-		return changeFilePermission(path, fsGroup, mounter.GetAttributes().ReadOnly, info)
+		return changeFilePermission(path, fsUser, fsGroup, mounter.GetAttributes().ReadOnly, info)
 	})
 }
 
-func changeFilePermission(filename string, fsGroup *int64, readonly bool, info os.FileInfo) error {
```

Comment:
the changes in this method need very close review by @kubernetes/sig-storage-pr-reviews 

/assign @jsafrane @msau42

Reply:
/assign @gnufied 
more eyes the better :-)

Reply:
Is this behavior protected by any feature gate? When we added the new fsgroup change policy behavior, we guarded the change to this method with a feature gate so we could disable it if it caused regressions to existing behavior.

Reply:
Regarding tests, we do have some testing fsgroup + projected volumes, but I don't think they cover the projected service account volume:

https://github.com/kubernetes/kubernetes/blob/master/test/e2e/common/projected_configmap.go#L59
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/common/projected_downwardapi.go
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/common/projected_secret.go

We don't have any e2es covering persistent volumes with runAsUser set (only fsgroup). And for fsgroup, it's only set in some basic tests like: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/testsuites/volumes.go#L177

I thought we had added fsgroup tests to subpath due to various regression issues, but I don't see fsgroup being set in those tests. I will dig further to figure out what happened to them.